packages/odata/src/services/grid-odata.service.ts

```diff
@@ -94,12 +94,70 @@ export class GridOdataService implements BackendService {
       const tmpColumnDefinitions = sharedService?.allColumns ?? grid.getColumns() ?? [];
       this._columnDefinitions = tmpColumnDefinitions.filter((column: Column) => !column.excludeFromQuery);
     }
+
+    this._odataService.columnDefinitions = this._columnDefinitions;
   }
 
   buildQuery(): string {
     return this._odataService.buildQuery();
   }
 
+  postProcess(processResult: any): void {
+    if (!processResult) {
```

Comment:
Actually, thinking about it. It can't be null/undefined at this point because of the check in `backendUtility.service.ts`:
```typescript
// allow the backend service to change the result.
    if (processResult && backendApi.service.postProcess) {
      backendApi.service.postProcess(processResult);
    }
```

So I'll just remove the check here.